writers/streamingbatchwriter/streamingbatchwriter.go

```diff
@@ -396,25 +397,27 @@ func (s *streamingWorkerManager[T]) run(ctx context.Context, wg *sync.WaitGroup,
 			}
 
 			var recSize int64
+			rowSize := int64(1) // at least 1 row for messages without records
 			if ins, ok := any(r).(*message.WriteInsert); ok {
 				recSize = util.TotalRecordSize(ins.Record)
+				rowSize = ins.Record.NumRows()
 			}
 
-			if (s.batchSizeRows > 0 && sizeRows >= s.batchSizeRows) || (s.batchSizeBytes > 0 && sizeBytes+recSize >= s.batchSizeBytes) {
+			ensureOpened()
+			clientCh <- r
```

Comment:
Same comment here about flushing after hitting the limit, not before

Reply:
9dbac8d613ad90520e9c2daf08314d3528c00a72